src/transformers/models/hiera/modeling_hiera.py

```diff
@@ -0,0 +1,1655 @@
+        w0 = width // self.patch_stride[1] if not self.sep_pos_embed else width // self.patch_stride[2]
+        # we add a small number to avoid floating point error in the interpolation
+        # see discussion at https://github.com/facebookresearch/dino/issues/8
+        h0, w0 = h0 + 0.1, w0 + 0.1
+        pos_embeds = pos_embeds.reshape(1, int(math.sqrt(num_positions)), int(math.sqrt(num_positions)), dim)
+        pos_embeds = pos_embeds.permute(0, 3, 1, 2)
+        pos_embeds = nn.functional.interpolate(
+            pos_embeds,
+            scale_factor=(h0 / math.sqrt(num_positions), w0 / math.sqrt(num_positions)),
+            mode="bicubic",
+            align_corners=False,
+        )
+        if int(h0) != pos_embeds.shape[-2] or int(w0) != pos_embeds.shape[-1]:
+            raise ValueError("The interpolated position encoding does not have the right size")
+        pos_embeds = pos_embeds.permute(0, 2, 3, 1).view(1, -1, dim)
+        return pos_embeds
+
+    def get_position_embedding(
+        self, embeddings: torch.Tensor, height: int, width: int, interpolate_pos_encoding: bool
+    ) -> torch.Tensor:
+        if self.sep_pos_embed:
+            spatial = self.position_embeddings_spatial
+            spatial = (
+                self.interpolate_pos_encoding(embeddings, spatial, height, width)
+                if interpolate_pos_encoding
+                else spatial
+            )
+            spatial = spatial.repeat(1, self.tokens_spatial_shape[0], 1)
+
+            temporal = torch.repeat_interleave(
+                self.position_embeddings_temporal,
+                self.tokens_spatial_shape[1] * self.tokens_spatial_shape[2],
+                dim=1,
+            )
+
+            return spatial + temporal
+        else:
+            position_embeddings = self.position_embeddings
+            position_embeddings = (
+                self.interpolate_pos_encoding(embeddings, position_embeddings, height, width)
+                if interpolate_pos_encoding
+                else position_embeddings
+            )
+            return position_embeddings
+
+    def forward(
+        self,
+        pixel_values: torch.Tensor,
+        noise: Optional[torch.FloatTensor] = None,
+        interpolate_pos_encoding: bool = False,
+    ) -> torch.Tensor:
+        if len(self.tokens_spatial_shape) == 2:
+            batch_size, num_channels, height, width = pixel_values.shape
+        else:
+            batch_size, num_channels, depth, height, width = pixel_values.shape
+
+        embeddings, mask, ids_restore = self.patch_embeddings(
+            pixel_values, noise=noise, interpolate_pos_encoding=interpolate_pos_encoding
+        )
+
+        embeddings = embeddings + self.get_position_embedding(embeddings, height, width, interpolate_pos_encoding)
+
+        return embeddings, mask, ids_restore
+
+
+class HieraMaskUnitAttention(nn.Module):
+    """
+    Computes either Mask Unit or Global Attention. Also is able to perform q pooling.
```

Comment:
```suggestion
    Computes either Mask Unit or Global Attention. Also is able to perform query pooling.
```
I assume this is query pooling